### PR TITLE
Boilerplate server

### PR DESCRIPTION
**Pull request by laptou**
- Create a Servlet server that runs on Java 11

You can run the server by just going into the `server` directory and running `run.sh`.

```bash
cd server
./run.sh
```

Then going to `http://localhost:8080/` should get you a "Hello, world!"

### CODE REVIEW

File: jetty/pom.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (laptou) [inline jetty/pom.xml:1]: Like I wrote in the README, this file is needed in order to create a simple server that will load our servlets and run them.

Comment (laptou) [reply jetty/pom.xml:1]: It was adapted from [here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/appengine-java11/appengine-simple-jetty-main/pom.xml).

File: jetty/src/main/java/com/google/musicanalysis/jettymain/Main.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (laptou) [inline jetty/src/main/java/com/google/musicanalysis/jettymain/Main.java:1]: This file was adapted from [here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/appengine-java11/appengine-simple-jetty-main/src/main/java/com/example/appengine/demo/jettymain/Main.java). It executes the WAR file, which is a type of archive that contains our Servlets.

File: server/pom.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.3.0</version>
+        <configuration>
+          <deploy.projectId>capstone-t99-2020</deploy.projectId>
+          <deploy.version>1</deploy.version>
+        </configuration>
+      </plugin>
+      <!-- Conforms Java code to Google style guide. 
+           $ mvn com.coveo:fmt-maven-plugin:format -->
+      <plugin>
```
Comment (laptou) [inline server/pom.xml:59]: Borrowed this from Seun's PR. +1 to Seun.

File: server/pom.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    </dependency>
+
+    <!-- Gson dependency -->
+    <dependency>
+        <groupId>com.google.code.gson</groupId>
+        <artifactId>gson</artifactId>
+        <version>2.8.6</version>
+    </dependency>
+  </dependencies>
+
```
Comment (seunomonije) [inline server/pom.xml:42]:
```suggestion

<dependency>
  <groupId>com.sachinhandiekar</groupId>
  <artifactId>jMusixMatch</artifactId>
  <version>1.1.4</version>
</dependency>

```
add this here for the musixmatch wrapper

File: jetty/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# jetty
+
+When running under App Engine with Java 11, we are required to have a file with a `main()` method in
+it for App Engine to call. In the Java 8 version, this was abstracted away because you could _only_
+use servlets, but Java 11 allows people to use frameworks other than servlets, so it's a little more
+complex.
+
+All the code in this folder does is replicate the functionality that App Engine was doing for us: it
+makes a dead-simple HTTP server that just hands off all of the work to our servlets. This means that
+the code in this folder is not likely to change over the course of this project.
```
Comment (seunomonije) [inline jetty/README.md:10]: +1 for the explanation

File: server/run.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+mvn clean package
+cd ../jetty
+mvn install
+mvn exec:java -Dexec.args="../server/target/site.war"
```
Comment (seunomonije) [inline server/run.sh:5]: +1 for creating a script to make it easier to run

File: jetty/README.md (hunk 5 above)
Comment (gracetian6) [reply jetty/README.md:10]: Java 11 finally catching on 👍

File: server/src/main/java/com/google/musicanalysis/site/IndexServlet.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/")
+public class IndexServlet extends HttpServlet {
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse res)
+      throws ServletException, IOException {
+    res.getWriter().write("Hello world!");
```
Comment (gracetian6) [inline server/src/main/java/com/google/musicanalysis/site/IndexServlet.java:15]: yesss 💯 💯

Comment (laptou) [reply server/src/main/java/com/google/musicanalysis/site/IndexServlet.java:15]: 😂

**Comment by seunomonije** [comment]
One thing to note as well with the server is that the user might have to run the script twice in order for the server to run properly. I was doing some debugging with a fresh repo and noticed the issue

File: jetty/src/main/java/com/google/musicanalysis/jettymain/Main.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+/** Simple Jetty Main that can execute a WAR file when passed as an argument. */
+public class Main {
+
+  public static void main(String[] args) throws Exception {
+    if (args.length != 1) {
+      System.err.println("Usage: need a relative path to the war file to execute");
+      System.exit(1);
+    }
+    System.setProperty("org.eclipse.jetty.util.log.class", "org.eclipse.jetty.util.log.StrErrLog");
```
Comment (geekster777) [inline jetty/src/main/java/com/google/musicanalysis/jettymain/Main.java:35]: nit: add a new line between the end of the if and the next chunk of code

**Pull request merged**